libs/chain-signatures/contract/src/state/running.rs

```diff
@@ -184,22 +194,27 @@ impl RunningContractState {
             Ok(None)
         }
     }
+
+    pub fn vote_code_hash(&mut self, code_hash: CodeHash) -> Result<Option<Self>, Error> {
```

Comment:
I am a bit confused about the return type of this function. It seems that it either returns an error or returns `Ok(None)`. In this case, why do we make the return type `Result<Option<Self>, Error>` instead of something like `Result<(), Error>`? It also makes the code that calls it look confusing